internal/queryexecutor/executor.go

```diff
@@ -0,0 +1,160 @@
+package queryexecutor
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+
+	"github.com/IBM-Blockchain/bcdb-server/internal/worldstate"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/constants"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/logger"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
+	"github.com/pkg/errors"
+)
+
+type WorldStateQueryExecutor struct {
+	db     worldstate.DB
+	logger *logger.SugarLogger
+}
+
+func NewWorldStateQueryExecutor(db worldstate.DB, l *logger.SugarLogger) *WorldStateQueryExecutor {
+	return &WorldStateQueryExecutor{
+		db:     db,
+		logger: l,
+	}
+}
+
+type attributesConditions map[string]*attrConditions
+
+type attrConditions struct {
+	valueType  types.Type
+	conditions map[string]interface{}
+}
+
+func (e *WorldStateQueryExecutor) validateAndDisectConditions(dbName string, conditions map[string]interface{}) (attributesConditions, error) {
+	// when we reach here, we assume that the given dbName exist
+	marshledIndexDef, _, err := e.db.GetIndexDefinition(dbName)
+	if err != nil {
+		return nil, err
+	}
+
+	if marshledIndexDef == nil {
+		return nil, errors.New("no index has been defined on the database " + dbName)
+	}
+
+	indexDef := map[string]types.Type{}
+	if err := json.Unmarshal(marshledIndexDef, &indexDef); err != nil {
+		return nil, err
+	}
```

Comment:
Currently, it is used in 2 packages. Further, only the following lines are duplicated in both
```
	if marshledIndexDef == nil {
		return nil, errors.New("no index has been defined on the database " + dbName)
	}

	indexDef := map[string]types.Type{}
	if err := json.Unmarshal(marshledIndexDef, &indexDef); err != nil {
		return nil, err
	}
```

Hence, I would hold on to it for now before moving this to a separate util package. Small amount of duplication is fine. 

Yes, when we introduce caching layer at the database, we can improve the performance. It will not be part of this PR.